Build failure: linuxPackages.ipu6-drivers:

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

build linuxPackages.ipu6-drivers: with kernel version >= 6.13

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/286363220

### Relevant log output

```console
Running phase: unpackPhase
unpacking source archive /nix/store/6zjjykxq8sm388l1ipdqx4n9z7fj2a0n-source
source root is source
Running phase: patchPhase
applying patch /nix/store/6zjjykxq8sm388l1ipdqx4n9z7fj2a0n-source/patches/0001-v6.10-IPU6-headers-used-by-PSYS.patch
patching file drivers/media/pci/intel/ipu6/ipu6-bus.h
patching file drivers/media/pci/intel/ipu6/ipu6-buttress.h
patching file drivers/media/pci/intel/ipu6/ipu6-cpd.h
patching file drivers/media/pci/intel/ipu6/ipu6-fw-com.h
patching file drivers/media/pci/intel/ipu6/ipu6-mmu.h
patching file drivers/media/pci/intel/ipu6/ipu6-platform-buttress-regs.h
patching file drivers/media/pci/intel/ipu6/ipu6-platform-regs.h
patching file drivers/media/pci/intel/ipu6/ipu6.h
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/backport-include' -> './backport-include'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/backport-include/drivers' -> './backport-include/drivers'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/backport-include/drivers/misc' -> './backport-include/drivers/misc'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/backport-include/drivers/misc/mei' -> './backport-include/drivers/misc/mei'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/backport-include/drivers/misc/mei/hbm.h' -> './backport-include/drivers/misc/mei/hbm.h'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/backport-include/drivers/misc/mei/hw.h' -> './backport-include/drivers/misc/mei/hw.h'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/backport-include/drivers/misc/mei/mei_dev.h' -> './backport-include/drivers/misc/mei/mei_dev.h'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/gpio' -> './drivers/gpio'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/gpio/gpio-ljca.c' -> './drivers/gpio/gpio-ljca.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/i2c' -> './drivers/i2c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/i2c/busses' -> './drivers/i2c/busses'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/i2c/busses/i2c-ljca.c' -> './drivers/i2c/busses/i2c-ljca.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/mfd' -> './drivers/mfd'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/mfd/ljca.c' -> './drivers/mfd/ljca.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc' -> './drivers/misc'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/ivsc' -> './drivers/misc/ivsc'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/ivsc/Kconfig' -> './drivers/misc/ivsc/Kconfig'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/ivsc/Makefile' -> './drivers/misc/ivsc/Makefile'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/ivsc/intel_vsc.c' -> './drivers/misc/ivsc/intel_vsc.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/ivsc/intel_vsc.h' -> './drivers/misc/ivsc/intel_vsc.h'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/ivsc/mei_ace.c' -> './drivers/misc/ivsc/mei_ace.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/ivsc/mei_ace_debug.c' -> './drivers/misc/ivsc/mei_ace_debug.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/ivsc/mei_csi.c' -> './drivers/misc/ivsc/mei_csi.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/ivsc/mei_pse.c' -> './drivers/misc/ivsc/mei_pse.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/mei' -> './drivers/misc/mei'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/mei/hw-vsc.c' -> './drivers/misc/mei/hw-vsc.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/mei/hw-vsc.h' -> './drivers/misc/mei/hw-vsc.h'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/misc/mei/spi-vsc.c' -> './drivers/misc/mei/spi-vsc.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/spi' -> './drivers/spi'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/drivers/spi/spi-ljca.c' -> './drivers/spi/spi-ljca.c'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/include/linux' -> './include/linux'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/include/linux/mfd' -> './include/linux/mfd'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/include/linux/mfd/ljca.h' -> './include/linux/mfd/ljca.h'
'/nix/store/yhl1fl2cjqy985liknijss85qj4pplzn-source/include/linux/vsc.h' -> './include/linux/vsc.h'
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
build flags: -j2 SHELL=/nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/bash KBUILD_OUTPUT=/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/build ARCH=x86_64 CROSS_COMPILE= KERNELRELEASE=6.13.0 KERNEL_SRC=/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/build
make -C /nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/build M=/build/source modules
make[1]: Entering directory '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/build'
make[2]: Entering directory '/build/source'
  CC [M]  drivers/media/i2c/hm11b1.o
  CC [M]  drivers/media/pci/intel/ipu6/psys/ipu-psys.o
drivers/media/pci/intel/ipu6/psys/ipu-psys.c: In function 'ipu6_psys_probe':
drivers/media/pci/intel/ipu6/psys/ipu-psys.c:2313:22: warning: assignment discards 'const' qualifier from pointer target type [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdiscarded-qualifiers-Wdiscarded-qualifiers8;;]
 2313 |         adev->auxdrv = to_auxiliary_drv(dev->driver);
      |                      ^
In file included from /nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/linux/module.h:22,
                 from /nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/linux/device/driver.h:21,
                 from /nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/linux/device.h:32,
                 from drivers/media/pci/intel/ipu6/psys/ipu-psys.c:6:
drivers/media/pci/intel/ipu6/psys/ipu-psys.c: At top level:
drivers/media/pci/intel/ipu6/psys/ipu-psys.c:2684:18: error: expected ',' or ';' before 'DMA_BUF'
 2684 | MODULE_IMPORT_NS(DMA_BUF);
      |                  ^~~~~~~
/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/linux/moduleparam.h:26:61: note: in definition of macro '__MODULE_INFO'
   26 |                 = __MODULE_INFO_PREFIX __stringify(tag) "=" info
      |                                                             ^~~~
/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/linux/module.h:299:33: note: in expansion of macro 'MODULE_INFO'
  299 | #define MODULE_IMPORT_NS(ns)    MODULE_INFO(import_ns, ns)
      |                                 ^~~~~~~~~~~
drivers/media/pci/intel/ipu6/psys/ipu-psys.c:2684:1: note: in expansion of macro 'MODULE_IMPORT_NS'
 2684 | MODULE_IMPORT_NS(DMA_BUF);
      | ^~~~~~~~~~~~~~~~
drivers/media/pci/intel/ipu6/psys/ipu-psys.c:2687:18: error: expected ',' or ';' before 'INTEL_IPU6'
 2687 | MODULE_IMPORT_NS(INTEL_IPU6);
      |                  ^~~~~~~~~~
/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/linux/moduleparam.h:26:61: note: in definition of macro '__MODULE_INFO'
   26 |                 = __MODULE_INFO_PREFIX __stringify(tag) "=" info
      |                                                             ^~~~
/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/include/linux/module.h:299:33: note: in expansion of macro 'MODULE_INFO'
  299 | #define MODULE_IMPORT_NS(ns)    MODULE_INFO(import_ns, ns)
      |                                 ^~~~~~~~~~~
drivers/media/pci/intel/ipu6/psys/ipu-psys.c:2687:1: note: in expansion of macro 'MODULE_IMPORT_NS'
 2687 | MODULE_IMPORT_NS(INTEL_IPU6);
      | ^~~~~~~~~~~~~~~~
make[5]: *** [/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/scripts/Makefile.build:194: drivers/media/pci/intel/ipu6/psys/ipu-psys.o] Error 1
make[4]: *** [/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/scripts/Makefile.build:440: drivers/media/pci/intel/ipu6/psys] Error 2
make[4]: *** Waiting for unfinished jobs....
  CC [M]  drivers/media/i2c/ov01a1s.o
  CC [M]  drivers/media/i2c/ov01a10.o
  CC [M]  drivers/media/i2c/ov02c10.o
  CC [M]  drivers/media/i2c/ov02e10.o
  CC [M]  drivers/media/i2c/ov05c10.o
  CC [M]  drivers/media/i2c/ov2740.o
  CC [M]  drivers/media/i2c/hm2170.o
  CC [M]  drivers/media/i2c/hm2172.o
  CC [M]  drivers/media/i2c/hi556.o
make[3]: *** [/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/Makefile:1989: .] Error 2
make[2]: *** [/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/Makefile:251: __sub-make] Error 2
make[2]: Leaving directory '/build/source'
make[1]: *** [/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/source/Makefile:251: __sub-make] Error 2
make[1]: Leaving directory '/nix/store/6b9j17qfxdvyh9kmi9fjgh6zg0ix8if5-linux-6.13-dev/lib/modules/6.13.0/build'
make: *** [Makefile:126: all] Error 2
```

### Additional context

An upstream patch is available at https://github.com/intel/ipu6-drivers/pull/321

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.0, NixOS, 25.05 (Warbler), 25.05.20250126.852ff1d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.25.4`
 - channels(root): `"nixos"`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`

### Notify maintainers


---

@flokli 

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/ipu6-drivers/default.nix
+++ b/pkgs/os-specific/linux/ipu6-drivers/default.nix
@@ -9,13 +9,13 @@
 
 stdenv.mkDerivation rec {
   pname = "ipu6-drivers";
-  version = "unstable-2024-11-19";
+  version = "unstable-2025-02-19";
 
   src = fetchFromGitHub {
     owner = "intel";
     repo = "ipu6-drivers";
-    rev = "0ad4988248d7e9382498a0b47fc78bb990b29a58";
-    hash = "sha256-UFvwuoAzwk1k4YiUK+4EeMKeTx9nVvBgBN5JKAfqZkQ=";
+    rev = "7af071481f3d2d3cef1e70113c10f62ac6351723";
+    hash = "sha256-pe7lqK+CHpgNWpC8GEZ3FKfYcuVuRUaWlW18D9AsrSk=";
   };
 
   patches = [

